File: backend/src/main/java/com/campussync/campussync_backend/dto/CreateStudentResponse.java
package com.campussync.campussync_backend.dto;

import lombok.AllArgsConstructor;
import lombok.Getter;

@Getter
@AllArgsConstructor
public class CreateStudentResponse {

    private Long userId;
    private String name;
    private String email;
    private String registerNumber;
    private String temporaryPassword;
}

File: frontend/lib/models/create_student_response.dart
class CreateStudentResponse {
  final int userId;
  final String name;
  final String email;
  final String registerNumber;
  final String temporaryPassword;

  const CreateStudentResponse({
    required this.userId,
    required this.name,
    required this.email,
    required this.registerNumber,
    required this.temporaryPassword,
  });

  factory CreateStudentResponse.fromJson(
    Map<String, dynamic> json,
  ) {
    return CreateStudentResponse(
      userId:
          (json['userId'] as num?)?.toInt() ?? 0,
      name:
          json['name']?.toString() ?? '',
      email:
          json['email']?.toString() ?? '',
      registerNumber:
          json['registerNumber']?.toString() ?? '',
      temporaryPassword:
          json['temporaryPassword']?.toString() ?? '',
    );
  }
}